src/app/share-preview/components/share-preview/share-preview.component.html

```diff
@@ -89,7 +89,13 @@
       </div>
     </div>
     <div class="share-preview-cover-buttons" (click)="stopPropagation($event)" *ngIf="isLoggedIn" >
-      <button class="btn btn-wordpress" (click)="onRequestAccessClick()" [disabled]="waiting || hasRequested" *ngIf="isLinkShare">{{ hasRequested ? 'Access Requested' : requestAccessText }}</button>
+      <button class="btn btn-wordpress" (click)="onRequestAccessClick()" [disabled]="waiting || hasRequested" *ngIf="isLinkShare">
+        <ng-container *ngIf="hasRequested ; else accessText">Access Requested</ng-container>
+        <ng-template #accessText>
+          <ng-container *ngIf="isAutoApprove ; else requestAccess">Accept Share</ng-container>
+          <ng-template #requestAccess>Request Access</ng-template>
+        </ng-template>
```

Comment:
This looks better, thank you!

If you're willing, I'd love to hear what you like about `ng-template`s! I think maybe I've over-internalized how Handlebars does things, because I would have defaulted to something more like this:

```
        <ng-container *ngIf="hasRequested">Access Requested</ng-container>
        <ng-container *ngIf="!hasRequested && isAutoApprove">Accept Share</ng-container>
        <ng-container *ngIf="!hasRequested && !isAutoApprove">Request Access</ng-container>>
```

I suppose maybe this construction leans on the developer to make sure they're correctly handling the if-else sequence, whereas the template approach ensures that only one of these will render?

Reply:
Yes, I'd prefer in this structure to have a specific `else` clause instead of having to specify with mutliple `if`s.

(This would be cleaner in something like Vue which has appropriate `v-if`, `v-else`, and `v-else-if` directives.)

Reply:
Thank you! That makes sense to me, and I appreciate the explanation.